### ISSUE TITLE
Web - Inconsistently highlighting occurs when multiple mentions are used without spaces.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21753]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1.	Go to any chat.
2.	Type "@ here" multiple times without including spaces.
3.	Observe that all the mentions are highlighted.
4.	Type a user's email ID followed by "@ here" without a space.
5.	Note that the email address is highlighted, but "@ here" is not.
## Expected Result: 
Consistency is expected, where both cases should either be highlighted or not.
## Actual Result: 
Multiple use of "@ here" are highlighted, but when using mentions with a user email followed by "@ here," it does not get highlighted.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.33-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/126239a3-a21b-474a-aaee-dc69411e9072

https://github.com/Expensify/App/assets/43996225/a79acc00-d07d-4b56-bbb4-614adba012e8

**Expensify/Expensify Issue URL:**
**Issue reported by:** @usmantariq96
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687759061501769

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Shivam7-1** [comment]
## Proposal

### LHN reflects report value after a click on LHN

## Please re-state the problem that we are trying to solve in this issue.
What is the root cause of that problem?
The issue occurs in `NewChatPage.js` File In this The error in the code lies in the `getSections` method. 
In the if condition that checks for `hasUnselectedUserToInvite`, the check is incorrect. It currently checks if `this.state.userToInvite` exists and if it is not included in the `filterText`. However, the `filterText` is constructed by concatenating the `login` property of selected options, so it will not include the email part of the userToInvite. T
his causes the `userToInvite` option to be consistently highlighted, even if it is selected.



### What changes do you think we should make in order to solve the problem?
In Solution of this issue The code can modify as follows:
To fix this issue, We need to modify the if condition in the `getSections` method as follows:


By checking `filterText.includes(this.state.userToInvite.email)` instead of `filterText.includes(this.state.userToInvite.login)`, we need to  ensure that the email part of `userToInvite` is included in the check.

This change will ensure that both cases (mentions without spaces and mentions with user email) are consistently highlighted or not highlighted based on their selection status

**Comment by flaviadefaria** [comment]
Added the external label.

**Comment by Shivam7-1** [comment]
## Proposal

### Inconsistently highlighting occurs when multiple mentions are used without spaces. 

## Please re-state the problem that we are trying to solve in this issue.
What is the root cause of that problem?
The issue occurs in `NewChatPage.js` File In this The error in the code lies in the `getSections` method. 
In the if condition that checks for `hasUnselectedUserToInvite`, the check is incorrect. It currently checks if `this.state.userToInvite` exists and if it is not included in the `filterText`. However, the `filterText` is constructed by concatenating the `login` property of selected options, so it will not include the email part of the userToInvite. T
his causes the `userToInvite` option to be consistently highlighted, even if it is selected.



### What changes do you think we should make in order to solve the problem?
In Solution of this issue The code can modify as follows:
To fix this issue, We need to modify the if condition in the `getSections` method as follows:


By checking `filterText.includes(this.state.userToInvite.email)` instead of `filterText.includes(this.state.userToInvite.login)`, we need to  ensure that the email part of `userToInvite` is included in the check.

This change will ensure that both cases (mentions without spaces and mentions with user email) are consistently highlighted or not highlighted based on their selection status

**Comment by flaviadefaria** [comment]
@aimane-chnaif thoughts on the proposal above?

**Comment by payal-lathidadiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Inconsistently highlighting occurs when multiple mentions are used without spaces

### What is the root cause of that problem?
When @here mention is added along side user mention, ExpensiMark parser doesn't parse `@here` mention correctly. The reason is below regular expression for `here-mention` rule [code here](https://github.com/Expensify/expensify-common/blob/main/lib/ExpensiMark.js#L95)
```
/[`.a-zA-Z]?@here(?=_\b|\b)(?!((?:(?!<a).)+)?<\/a>|[^<]*(<\/pre>|<\/code>))/gm
```
This regex doesn't extract correct token `@here` from match result [test link](https://regex101.com/r/FootjZ/1) to replace with our user-mention markdown.

However, with `@here@here` comment it extracts, `@here` match token correctly [test link](https://regex101.com/r/FUThIm/1)

Hence, this caused inconsistency in highlight of here mention with multiple mentions.

### What changes do you think we should make in order to solve the problem?
To fix here mention identification when mentioned with other user mention, we can change regular expression with below [code here](https://github.com/Expensify/expensify-common/blob/main/lib/ExpensiMark.js#L95)
```
/@here(?=_\b|\b)(?!((?:(?!<a).)+)?<\/a>|[^<]*(<\/pre>|<\/code>))/gm
```
[Test link with updated regex](https://regex101.com/r/qXXFzI/1)

### What alternative solutions did you explore? (Optional)
None

**Comment by flaviadefaria** [comment]
@aimane-chnaif thoughts on the proposals above?

**Comment by Shivam7-1** [comment]
> @Shivam7-1 The root cause is not correct

Okay I will check again and make changes in proposal accordingly

**Comment by ShogunFire** [comment]
Hello @aimane-chnaif What about @payal-lathidadiya proposal ?
I tested it and it looks like it is working

**Comment by payal-lathidadiya** [comment]
> @payal-lathidadiya can you please explain in detail about your solution (regex)?

In current regex, we have `[.a-zA-Z]?` in starting before `@here`. So with `?` specifier, it will match the previous token between zero and one times. So, while parsing here mention token with this regex, it will extract match result as `<previous-char>@here` when `@here` is added along side user mention which doesn't get validated [here](https://github.com/Expensify/expensify-common/blob/main/lib/ExpensiMark.js#L97)

My proposed solution suggests to remove this initial part `[.a-zA-Z]?` from regex since we only want to match whole word `@here` from input

**Comment by flaviadefaria** [comment]
@aimane-chnaif what are your thoughts on the latest proposals?

**Comment by flaviadefaria** [comment]
Doubled the bounty to see if we get more proposals.

**Comment by ShogunFire** [comment]
Can you confirm this is the expected behaviour for your examples ?

```

@aiman.chnaif+2@gmail.com@here => <m>@aiman.chnaif+2@gmail.com</m><m>@here</m>

@here@aimaine.chnaif+2@gmail.com => <m>@here</m><m>@aimaine.chnaif+2@gmail.com</m>

@here@here => <m>@here</m><m>@here<m>

@aiman.chnaif+2@gmail.com@aiman.chnaif+2@gmail.com => 
<m>@aiman.chnaif+2@gmail.com</m><m>@aiman.chnaif+2@gmail.com</m>

@here@here@gmail.com => <m>@here</m><m>@here</m>@gmail.com
@here@gmail.com@here => <m>@here</m>@gmail.com<m>@here</m>
@here@here+1@gmail.com => <m>@here</m><m>@here+1</m>@gmail.com
@here+1@gmail.com@here => <m>@here</m>+1@gmail.com<m>@here</m>

```

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When we have two mentions following eachothers the highlights are insconsistent

### What is the root cause of that problem?

As explained in other proposals the @here mention regexp is wrong

### What changes do you think we should make in order to solve the problem?

If we always want @here mention to be highlighted "alone" and not part of an email, I think we can just look for @here and surround it with `<mention-here>`
The code would look like this:
```
{
    name: 'hereMentions',
    regex: /@here/gm,
    replacement: (match) => {
        return `<mention-here>${match}</mention-here>`;
    },
},

```
It looks too simple so I don't know if this is what we want ?

Result:

![image](https://github.com/Expensify/App/assets/19537677/6641f00f-c1e8-444a-8017-d6ffbd6b6e7f)





### What alternative solutions did you explore? (Optional)

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

When we have multiple mention following eachother, the highlight is wrong 

### What is the root cause of that problem?

The here mention regexp is wrong and doesn't take into account every cases.

### What changes do you think we should make in order to solve the problem?
Please bear with me, this regexp seems hard so I am going to try to explain:
I think we should change the here mention regexp by:
```/(@here)([a-zA-Z0-9.!$%&+/=?^\`{|}_-]?)(?=\b)(?!(([\w\-\+\'#]*(?:\\.[\w\-\'\+]+)*@(?:[\w\-]+\.)+[a-z]{2,}(?!(([\w\-\+\'#\.\\]*(?:\\.[\w\-\'\+]+)*@(?:[\w\-]+\.)+[a-z]{2,}))))|((?:(?!<a).)+)?<\/a>|[^<]*(<\/pre>|<\/code>)))/gm,```

Result: 
![image](https://github.com/Expensify/App/assets/19537677/1311dd19-f3d1-40f0-85cb-db63c52a80ed)

Explanation:
We want to prevent matching @here@gmail.com and @here+1@gmail.com because here@gmail.com and here+1@gmail.com  are email addresses

So I first tested that the string after @here is not an email adress using the regexp found here : https://github.com/Expensify/expensify-common/blob/d7e424913550349acc779d029319af5b074c7c9e/lib/CONST.jsx#L3

But it wasn't working because @gmail.com is not an email adress so @here@gmail.com was still matching

So I modified it to accept @gmail.com as a valid email adress (I just replace a '+' by a '*' )
It was almost working:
![image](https://github.com/Expensify/App/assets/19537677/f5caf466-07a5-45ca-a359-2846f1c4a7cc)


Unfortunely it doesn't work if the email adress that is after @here has a dot. you can see that @here@aimaine.chnaif+2@gmail.com doesn't match because the regexp recognize here@aimaine.chnaif as an email (and it is)
So to prevent that we need to test that the part that would make an email with here is not part of another email:
So we had: ```(?!(([\w\-\+\'#\.\\]*(?:\\.[\w\-\'\+]+)*@(?:[\w\-]+\.)+[a-z]{2,}))) ```


For more clarity we can use variables instead of this big rexexp

### What alternative solutions did you explore? (Optional)

**Comment by Antasel** [comment]
# Proposal
### Please re-state the problem that we are trying to solve in this issue.
when user-mention and here-mention are followed by each other, highlight is wrong

### What is the root cause of that problem?
here-mention rule's regex is not covering the edge cases, need to fix it

### What changes do you think we should make in order to solve the problem?
Frist, here mention's regex pattern would be as following
```new RegExp(`([^@*~]?)((?:@${CONST.REG_EXP.EMAIL_PART})?)(@here)((?:@${CONST.REG_EXP.EMAIL_PART})?)([^@*~]?)(?=\\b)(?!([\\w'#%+-]*\\@(?:[a-z\\d-]+\\.)+[a-z]{2,})|((?:(?!<a).)+)?<\\/a>|[^<]*(<\\/pre>|<\\/code>))`, 'gm')```

Second, replacement logic has some tweaks according to regex pattern amendment
```
                replacement: (match, g1, g2, g4, g5, g7) => {
                    if (!Str.isValidMention(match) && g2 == '' && g5 == '') {
                        return match;
                    }
                    return `${g1}${g2}<mention-here>${g4}</mention-here>${g5}${g7}`;
                },
```

Explanation:
- g2, g5 is possible user mention captured by `((?:@${CONST.REG_EXP.EMAIL_PART})?)`, if existed, `@here` is parsed as here mention
- (?!([\\w'#%+-]*\\@(?:[a-z\\d-]+\\.)+[a-z]{2,})) prevents `@here@gmail.com` matched as here mention

With my approach, @aimane-chnaif 's edge cases are passed successfully
<details>
  <summary>Test cases</summary>

```
test('Test user mention and here mention, which are concatenated without space', () => {
    let testString = '@aiman.chnaif+2@gmail.com@here';
    let resultString = '<mention-user>@aiman.chnaif+2@gmail.com</mention-user><mention-here>@here</mention-here>';
    expect(parser.replace(testString)).toBe(resultString);

    testString = '@here@aiman.chnaif+2@gmail.com';
    resultString = '<mention-here>@here</mention-here><mention-user>@aiman.chnaif+2@gmail.com</mention-user>';
    expect(parser.replace(testString)).toBe(resultString);

    testString = '@aiman.chnaif+2@gmail.com@aiman.chnaif+2@gmail.com';
    resultString = '<mention-user>@aiman.chnaif+2@gmail.com</mention-user><mention-user>@aiman.chnaif+2@gmail.com</mention-user>';
    expect(parser.replace(testString)).toBe(resultString);

    testString = '@here@here@gmail.com';
    resultString = '<mention-here>@here</mention-here><mention-user>@here@gmail.com</mention-user>';
    expect(parser.replace(testString)).toBe(resultString);

    testString = '@here@gmail.com@here';
    resultString = '<mention-user>@here@gmail.com</mention-user><mention-here>@here</mention-here>';
    expect(parser.replace(testString)).toBe(resultString);

    testString = '@here@here+1@gmail.com';
    resultString = '<mention-here>@here</mention-here><mention-user>@here+1@gmail.com</mention-user>';
    expect(parser.replace(testString)).toBe(resultString);

    testString = '@here+1@gmail.com@here';
    resultString = '<mention-user>@here+1@gmail.com</mention-user><mention-here>@here</mention-here>';
    expect(parser.replace(testString)).toBe(resultString);
});
```

</details>
and it does not make other auto test cases failed
![image](https://github.com/Expensify/App/assets/139204153/cb818d22-111f-4ab9-bb58-5084d834a081)

you can check on [this branch](https://github.com/Antasel/expensify-common/tree/fix-here-mention)
cc: @aimane-chnaif

**Comment by Antasel** [comment]
Contributor details
Your Expensify account email: anayetrasel@yahoo.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01c94cf929c57ebc60?viewMode=1

**Comment by ShogunFire** [comment]
I agree that there are probably some cases that are not handled by my proposal but I have the general idea, more tests would have been done during the PR. Also I am not sure we can trust the current unit tests, for example there are weird cases like this:

@here_@here_.com

.According to the current unit  test that should return 

`<mention-here>@here</mention-here><em><mention-here>@here</mention-here></em>.com`

but isn't here_@here_.com a valid email adress ? I think it is (and github seem to agree)

**Comment by payal-lathidadiya** [comment]
> Awaiting proposals. No one proposed right solution to cover all these cases:
> 
> <img alt="Screenshot 2023-07-18 at 8 13 18 AM" width="409" src="https://user-images.githubusercontent.com/96077027/254173380-c5144fd5-8ed0-4b1f-8067-de507a0216ff.png">

@aimane-chnaif my proposal [here](https://github.com/Expensify/App/issues/21753#issuecomment-1635805462) seems working correctly for expected output.
I just checked and re-tested my solution with above given test cases, they are parsed correctly and validated.

Let me know if I'm missing anything.

**Comment by ShogunFire** [comment]
@payal-lathidadiya This image that @aimane-chnaif  put is not the expected output

![image](https://github.com/Expensify/App/assets/19537677/1dfe8395-c25d-4dec-89a3-4daba92872e6)

**Comment by Antasel** [comment]
> but isn't [here_@here_.com](mailto:here_@here_.com) a valid email adress ? I think it is (and github agrees)

@ShogunFire it's not valid email address. here_.com is not allowed , since underscore is not valid for domain level.

**Comment by ShogunFire** [comment]
@Antasel then our current regexp for email validation is wrong...


This is a headache to handle every cases...

@toto@here.com -> mention user
toto@here.com -> website
toto@heretutu -> here mention
@toto@gmail.com@here -> mention user and mention here

If @Antasel proposal don't handle all those cases I think we should just decide to only display the here mention when @here is surrounded by spaces,

**Comment by Antasel** [comment]
@ShogunFire I agree that current email validation regex is wrong, it would be fixed with #17387 , I also proposed a solution to cover all cases there, waiting for @aimane-chnaif 's review. But current my solution here would cover all cases even though email regex is fixed from #17387 

@aimane-chnaif could you please confirm final solution in #17387 ?

**Comment by ShogunFire** [comment]
My proposal doesn't work for every cases and I don't know how to make it work so I am just going to hide it :( good luck sir

**Comment by flaviadefaria** [comment]
@aimane-chnaif what's the latest here? Do any of the proposals work?

**Comment by flaviadefaria** [comment]
@aimane-chnaif there are quite a few updated proposals shared here. When do you think you can review them?

**Comment by flaviadefaria** [comment]
@aimane-chnaif can you give an update here?

**Comment by flaviadefaria** [comment]
@aimane-chnaif please give an update today.

**Comment by flaviadefaria** [comment]
@aimane-chnaif do you expect https://github.com/Expensify/App/issues/17387 to address this issue? It's not clear why we'd put this on hold for that issue, mind clarifying?

Also it might be worth removing some of the edge cases from this issue and focusing on what is most likely to happen?

**Comment by Antasel** [comment]
@aimane-chnaif 
I've checked that my previous solution does not works after email base regex is changed from #17387 , 
so i added new solution as alternative option which works even after the email regex is changed
Proposal is updated [here](https://github.com/Expensify/App/issues/21753#issuecomment-1643474183)

**Comment by Antasel** [comment]
I've figured out that look behind is not supported on Safari version less than 16.4 .
so I did some tweak on new hereMentionAfterUserMentions rule as alternative of lookbehind (?<=)
Alternative solution has been updated [here](https://github.com/Expensify/App/issues/21753#issuecomment-1643474183)

**Comment by Antasel** [comment]
@flaviadefaria @aimane-chnaif 
#17387 's PR has been merged. Could you please check the alternative solution on my proposal ?

**Comment by flaviadefaria** [comment]
@aimane-chnaif can you check @Antasel's comment above? Thanks!

**Comment by Antasel** [comment]
@flaviadefaria @aimane-chnaif 
Looks like merge freeze was over. could you check my proposal ? alternative solution

**Comment by flaviadefaria** [comment]
@aimane-chnaif I've removed the hold from the title can you please give an update here?

**Comment by flaviadefaria** [comment]
@aimane-chnaif should I reassign this?

**Comment by agilejune** [comment]
@flaviadefaria @aimane-chnaif 
@Antasel and I worked as team. Is it possible for me to contribute here ? here proposal was by me...

**Comment by agilejune** [comment]
no , but i can use my account and upwork account

**Comment by agilejune** [comment]
@aimane-chnaif  Proposal was by me, so I was interested in how you evalute about it. so opening this issue tab on my PC

**Comment by flaviadefaria** [comment]
@aimane-chnaif what do you recommend we do here? Continue looking for proposals,  have @agilejune re-submit the proposal from his account or something else? Also happy to move this to Slack if better.

**Comment by flaviadefaria** [comment]
@puneetlath I'm officially no longer in the BugZero team so reassigning this as it will not close until tomorrow.

**Comment by agilejune** [comment]
@aimane-chnaif 
How do you evaluate the [alternative solution](https://github.com/Expensify/App/issues/21753#issuecomment-1643474183) ? 
If it's acceptable, Can I re-submit it using my account ?

**Comment by agilejune** [comment]
@aimane-chnaif 
I don't have other proposal,  I just asked if i can contribute with Antasel's alternative solution, I explained it why.

**Comment by agilejune** [comment]
@aimane-chnaif 
do you think the proposal covers this issue correctly ?

**Comment by agilejune** [comment]
# Proposal
### Please re-state the problem that we are trying to solve in this issue.
when user-mention and here-mention are followed by each other, highlight is wrong

### What is the root cause of that problem?
here-mention rule's regex is not covering the edge cases, need to fix it

### What changes do you think we should make in order to solve the problem?
First, in here mention's regex pattern, add a tweak about negative lookbehind to prevent first `@here` in `@here+1@gmail.com`, `@here@gmail.com@here` matched as here mention
changed part would be like following:
```
(?!([\w'#%+-]*@(?:[a-z\d-]+\.)+[a-z]{2,}(?:\s|$|@here))|
```

Secondly, to match `@here` which is preceding or following the user mention, we need to add another rule like 'hereMentionAfterUserMentions' after userMentions rule
```
            {
                name: 'hereMentionAfterUserMentions',
                regex: /(<\/mention-user>)(@here)(?=\b)/gm,
                replacement: '$1<mention-here>$2</mention-here>',
            },
```

lastly, we can replace the long string [a-zA-Z0-9.!$%&+/=?^{|}_-] with [^@*~] in here mention's regex, it does not effect the result. so final here mention regex would be as following:
```
regex: /([^@*~]?)(@here)([^@*~]?)(?=\b)(?!([\w'#%+-]*@(?:[a-z\d-]+\.)+[a-z]{2,}(?:\s|$|@here))|((?:(?!<a).)+)?<\/a>|[^<]*(<\/pre>|<\/code>))/gm,
```

I tested all edge cases of main solution and does not make other auto test cases failed.

**Comment by puneetlath** [comment]
@aimane-chnaif bump on review!

**Comment by puneetlath** [comment]
Sounds good to me. Let's make sure we update the tests to cover the new scenarios.

**Comment by agilejune** [comment]
@puneetlath @aimane-chnaif 
The PR https://github.com/Expensify/expensify-common/pull/594 for expensify-common is ready

**Comment by puneetlath** [comment]
Great! @aimane-chnaif will take the first round of review.

**Comment by puneetlath** [comment]
@aimane-chnaif how's the review going?

**Comment by puneetlath** [comment]
@agilejune I merged the expensify-common PR. Please open a PR in Expensify/App to update the expensify-common version. Thanks!

**Comment by agilejune** [comment]
@puneetlath @aimane-chnaif 
The App PR https://github.com/Expensify/App/pull/31541 is ready

**Comment by puneetlath** [comment]
Ok so is there anything left to do here? Or it sounds like we're ready to pay out?

**Comment by puneetlath** [comment]
kk. @aimane-chnaif I sent you an offer. 

@agilejune can you please apply here? https://www.upwork.com/jobs/~01f57f05c12d9a825a

**Comment by puneetlath** [comment]
@agilejune bump. Can you please apply to the Upwork job so that we can close this out? Actually, also @usmantariq96. https://www.upwork.com/jobs/~01f57f05c12d9a825a

@aimane-chnaif has been paid

**Comment by usmantariq96** [comment]
@puneetlath link is not opening at my end.

Can you please send me direct message please.
Thanks

**Comment by puneetlath** [comment]
@agilejune to this one? https://www.upwork.com/jobs/~01f57f05c12d9a825a

I'm not seeing your application.

**Comment by agilejune** [comment]
@puneetlath 
![image](https://github.com/Expensify/App/assets/86615752/14dd8be4-8b47-484c-b2ad-144cc5aae1f7)

can you check this proposal ?

**Comment by agilejune** [comment]
> @agilejune to this one? https://www.upwork.com/jobs/~01f57f05c12d9a825a
> 
> I'm not seeing your application.

Yes, I applied there

**Comment by puneetlath** [comment]
Super weird. I don't see it. 

<img width="1376" alt="image" src="https://github.com/Expensify/App/assets/1371996/a51eb95c-5791-4c6a-a9fe-799e898761a3">

Can you message me your upwork profile either here or in Slack and I'll try inviting you directly. 

Also @usmantariq96 friendly bump to apply.

**Comment by agilejune** [comment]
@puneetlath 
this is my profile

https://www.upwork.com/freelancers/~017f0481af079ea44f

**Comment by agilejune** [comment]
I thought I saved my details to contribute here.
But let me try again 

Contributor details
Your Expensify account email: mokinmaksym@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~017f0481af079ea44f

**Comment by puneetlath** [comment]
Ok, offers sent: 

https://www.upwork.com/nx/wm/offer/27999511
https://www.upwork.com/nx/wm/offer/27999485

**Comment by puneetlath** [comment]
Ok only @agilejune left to pay once they accept.

**Comment by agilejune** [comment]
@puneetlath 
Thanks for your offer,
I've accepted
